src/support/editor.ts

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs'
+import { spawnSync } from 'child_process'
+
+// tslint:disable-next-line
+const tmp = require('tmp')
+
+export interface EditorParams {
+  json?: boolean;
+}
+
+export class Editor {
+  private _content: string;
+
+  private tmpFile: any;
+
+  private defaultEditor: string;
+
+  private options: EditorParams;
+
+  constructor(content = '', options: EditorParams = {}) {
+    this._content = content
+
+    this.options = options
+
+    this.defaultEditor = process.platform.startsWith('win')
+      ? 'notepad'
+      : 'nano'
+
+    if (this.options.json) {
+      if (this._content.length === 0) {
+        this._content = '{\n\n}'
+      }
+      else {
+        this._content = JSON.stringify(JSON.parse(this._content), null, 2)
+      }
+
+      this.tmpFile = tmp.fileSync({ postfix: '.json' })
+    }
+    else {
+      this.tmpFile = tmp.fileSync({ postfix: '.json' })
```

Comment:
Superfluous else branch: the tmpFile variable content doesn't depend on the result of the if condition

Reply:
Bad copy/paste ..